tests/test_utils.py

```diff
@@ -196,6 +262,63 @@ def test_sub_pixel_read(_source_image):
     sub_pixel_read(test_image, pillow_test_image, bounds, ow, oh)
 
 
+def test_aligned_padded_sub_pixel_read():
+    """Test sub-pixel numpy image reads with pixel-aligned bounds."""
+    image_path = Path(__file__).parent / "data" / "source_image.png"
+    assert image_path.exists()
+    test_image = utils.misc.imread(image_path)
+
+    x = 1
+    y = 1
+    w = 5
+    h = 5
+    padding = 1
+    bounds = (x, y, x + w, y + h)
+    ow = 4 + 2 * padding
+    oh = 4 + 2 * padding
+    output = utils.image.sub_pixel_read(test_image, bounds, (ow, oh), padding=padding)
+    assert (ow + 2 * padding, oh + 2 * padding) == tuple(output.shape[:2][::-1])
+
+
+def test_non_aligned_padded_sub_pixel_read():
+    """Test sub-pixel numpy image reads with non-pixel-aligned bounds."""
+    image_path = Path(__file__).parent / "data" / "source_image.png"
+    assert image_path.exists()
+    test_image = utils.misc.imread(image_path)
+
+    x = 0.5
+    y = 0.5
+    w = 4.5
+    h = 4.5
+    padding = 1
+    bounds = (x, y, x + w, y + h)
+    ow = 4
+    oh = 4
+    output = utils.image.sub_pixel_read(test_image, bounds, (ow, oh), padding=padding)
+
+    assert (ow + 2 * padding, oh + 2 * padding) == tuple(output.shape[:2][::-1])
+
+
+def test_non_baseline_padded_sub_pixel_read():
+    """Test sub-pixel numpy image reads with baseline padding."""
+    image_path = Path(__file__).parent / "data" / "source_image.png"
```

Comment:
Please use `_source_image`

Reply:
Fixed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/8b687c59ac3e9b5d2fc296ded0cae17e148670d3